test/Serilog.Sinks.Splunk.CustomFieldsTests/Serilog.Sinks.Splunk.CustomFieldsTests/SplunkCustomFieldsTests.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using NUnit.Framework;
+using Serilog.Events;
+using Serilog.Parsing;
+using Newtonsoft.Json;
+
+namespace Serilog.Sinks.Splunk.CustomFieldsTests
+{
+
+    // http://json2csharp.com/#
+    // https://github.com/JamesNK/Newtonsoft.Json
+    public class Event
+    {
+        public string Level { get; set; }
+        public string MessageTemplate { get; set; }
+        public string RenderedMessage { get; set; }
+        public string Exception { get; set; }
+    }
+    public class  TestCustomFields { 
+        public string  RelChan { get; set; }
+        public string Version { get; set; }
+        public string Rel { get; set; }
+        public List<string> Role { get; set; }
+    }
+    public class TestEventResultObject
+    {
+        public string Time { get; set; }
+        public Event @Event { get; set; }
+        public string Source { get; set; }
+        public string Sourcetype { get; set; }
+        public string Host { get; set; }
+        public string Index { get; set; }
+        public TestCustomFields Fields { get; set; }
+    }
+    [TestFixture]
+    class SplunkCustomFieldsTests
```

Comment:
There area some great use cases here, can we move to the existing test project.  We are keen to stay with XUnit due to a number of issues with NUnit & DotNet core.

As for the Arrange/Act/Assert.  We should be able to achieve in XUnit